pkg/distributors/test/Exiter.test.ts

```diff
@@ -0,0 +1,98 @@
+import { ethers } from 'hardhat';
+import { Contract, utils } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import Token from '@balancer-labs/v2-helpers/src/models/tokens/Token';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+
+import { bn, fp } from '@balancer-labs/v2-helpers/src/numbers';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { expectBalanceChange } from '@balancer-labs/v2-helpers/src/test/tokenBalance';
+import { advanceTime } from '@balancer-labs/v2-helpers/src/time';
+import { setup, rewardsDuration } from './MultiRewardsSharedSetup';
+
+describe('Exiter', () => {
+  let lp: SignerWithAddress, mockAssetManager: SignerWithAddress;
+
+  let poolTokens: TokenList;
+  let vault: Contract;
+  let stakingContract: Contract;
+  let callbackContract: Contract;
+  let rewardToken: Token;
+  let pool: Contract;
+
+  before('deploy base contracts', async () => {
+    [, , lp, mockAssetManager] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('set up asset manager and exiter', async () => {
+    const { contracts } = await setup();
+
+    pool = contracts.pool;
+    vault = contracts.vault;
+    stakingContract = contracts.stakingContract;
+    rewardToken = contracts.rewardTokens.DAI;
+    poolTokens = contracts.tokens;
+
+    callbackContract = await deploy('Exiter', { args: [vault.address] });
+  });
+
+  describe('with a stake and a reward', () => {
+    const rewardAmount = fp(1);
+    let assets: string[];
+    let poolId: string;
+
+    sharedBeforeEach(async () => {
+      await stakingContract
+        .connect(mockAssetManager)
+        .allowlistRewarder(pool.address, rewardToken.address, mockAssetManager.address);
+      await stakingContract.connect(mockAssetManager).addReward(pool.address, rewardToken.address, rewardsDuration);
+
+      const bptBalance = await pool.balanceOf(lp.address);
+
+      await pool.connect(lp).approve(stakingContract.address, bptBalance);
+
+      await stakingContract.connect(lp)['stake(address,uint256)'](pool.address, bptBalance);
```

Comment:
I know that Nico really isn't a fan of overloading function names because of this. Have you talked about possibly renaming the three arg version `stakeFor` or something?

Reply:
Seems reasonable, though fwiw I see this as a problem with ethers, not a problem with the choice to overload.  Made the change